src/v/utils/container_utils.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/coroutine/maybe_yield.hh>
+
+template<typename Container>
+concept IterableContainer
+  = requires(Container c, typename Container::iterator i) {
+    { c.begin() } -> std::same_as<typename Container::iterator>;
+    { c.end() } -> std::same_as<typename Container::iterator>;
+    { ++i } -> std::convertible_to<typename Container::iterator>;
+};
+
+template<typename Container>
+concept RangeErasableContainer = requires(
+  Container c, typename Container::iterator i, typename Container::iterator j) {
+    { c.erase(i, j) } -> std::same_as<typename Container::iterator>;
+};
+
+template<typename Container>
+concept SizableContainer = requires(Container c) {
+    { c.size() } -> std::convertible_to<size_t>;
+};
+
+template<typename Container>
+concept ClearableContainer = requires(Container c) {
+    c.clear();
+};
+
+template<typename Container>
+concept AsyncClearRequirements
+  = IterableContainer<Container> && RangeErasableContainer<
+    Container> && SizableContainer<Container> && ClearableContainer<Container>;
+
+/// Clears the given container while yielding to seastar after every
+/// element is erased. This is used to avoid reactor stall warnings
+/// when deleting containers with a large number of objects.
+///
+/// \param c a container to be erased.
+template<typename Container>
+ss::future<>
+async_clear(Container& c, const size_t threshold_size = 1000) requires
+  AsyncClearRequirements<Container> {
```

Comment:
Any reason not to use the nice syntax?:
```suggestion
template<AsyncClearRequirements Container>
ss::future<> async_clear(Container& c, const size_t threshold_size = 1000) {
```

Reply:
No reason beyond me forgetting those options exist lol. The abbreviated function template looks a lot better than what I was doing.